src/Text/Text.js

```diff
@@ -1,19 +1,68 @@
 import React, { forwardRef } from 'react'
 import classNames from 'classnames'
 
-// this component is not fully fleshed out, it's just here to add some typography
-function Text({ className, as, children }, ref) {
-  const Component = as ?? 'span'
+// this is the variant approach to text
+function Text({ as, variant, children }, ref) {
+  const Component = as ?? getHTMLElement(variant)
+
+  var className
```

Comment:
Nit: prefer let to var. We don't have a lint rule about this, so maybe we should, though I don't want to be overly pedantic if someone has a good reason to use var.